apps/language_server/lib/language_server/providers/execute_command/manipulate_pipes.ex

```diff
@@ -0,0 +1,308 @@
+
+    {:ok, call, range(line, col, end_line, end_col)}
+  end
+
+  defp do_get_function_call(text, start_char, end_char) do
+    text
+    |> do_get_function_call(start_char, end_char, %{paren_count: 0, text: ""})
+    |> Map.get(:text)
+    |> IO.iodata_to_binary()
+  end
+
+  defp do_get_function_call(<<c::binary-size(1), tail::bitstring>>, start_char, end_char, acc)
+       when c == start_char do
+    do_get_function_call(tail, start_char, end_char, %{
+      acc
+      | paren_count: acc.paren_count + 1,
+        text: [acc.text | [c]]
+    })
+  end
+
+  defp do_get_function_call(<<c::binary-size(1), tail::bitstring>>, start_char, end_char, acc)
+       when c == end_char do
+    acc = %{acc | paren_count: acc.paren_count - 1, text: [acc.text | [c]]}
+
+    if acc.paren_count <= 0 do
+      acc
+    else
+      do_get_function_call(tail, start_char, end_char, acc)
+    end
+  end
+
+  defp do_get_function_call(<<c::binary-size(1), tail::bitstring>>, start_char, end_char, acc) do
+    do_get_function_call(tail, start_char, end_char, %{acc | text: [acc.text | [c]]})
+  end
+
+  defp get_pipe_call(line, col, head, current, tail) do
+    pipe_right = do_get_function_call(tail, "(", ")")
+
+    pipe_left =
+      head
+      |> String.reverse()
+      |> :unicode.characters_to_binary(:utf8, :utf16)
+      |> do_get_pipe_call()
+      |> :unicode.characters_to_binary(:utf16, :utf8)
+
+    pipe_left =
+      if String.contains?(pipe_left, ")") do
+        get_function_call_before(head)
+      else
+        pipe_left
+      end
+
+    pipe_left = String.trim_leading(pipe_left)
+
+    pipe_call = pipe_left <> current <> pipe_right
+
+    {line_offset, tail_length} =
+      pipe_left
+      |> String.reverse()
+      |> count_newlines_and_get_tail()
+
+    start_line = line - line_offset
+
+    start_col =
+      if line_offset != 0 do
+        head
+        |> String.trim_trailing(pipe_left)
+        |> String.split(["\r\n", "\n", "\r"])
+        |> Enum.at(-1, "")
+        |> String.length()
+      else
+        col - tail_length
+      end
+
+    {line_offset, tail_length} = (current <> pipe_right) |> count_newlines_and_get_tail()
+
+    end_line = line + line_offset
+
+    end_col =
+      if line_offset != 0 do
+        tail_length
+      else
+        col + tail_length
+      end
+
+    {:ok, pipe_call, range(start_line, start_col, end_line, end_col)}
+  end
+
+  # do_get_pipe_call(text :: utf16 binary, {utf16 binary, has_passed_through_whitespace, should_halt})
+  defp do_get_pipe_call(text, acc \\ {"", false, false})
+
+  defp do_get_pipe_call(_text, {acc, _, true}), do: acc
+  defp do_get_pipe_call("", {acc, _, _}), do: acc
+
+  defp do_get_pipe_call(<<?\r::utf16, ?\n::utf16, _::bitstring>>, {acc, true, _}),
+    do: <<?\r::utf16, ?\n::utf16, acc::bitstring>>
+
+  defp do_get_pipe_call(<<0, c::utf8, _::bitstring>>, {acc, true, _})
+       when c in [?\t, ?\v, ?\r, ?\n, ?\s],
```

Comment:
is anyone still using vertical tabs?

Reply:
I've never encountered one in the wild haha